Projects/App/Sources/Journeys/FeaturesLoader.swift

```diff
@@ -83,3 +83,23 @@ struct NotificationLoader: Presentable {
         )
     }
 }
+
+extension UNUserNotificationCenter {
+    func startClaimIfStatusDeterminated(forOriginId originId: String) -> UNAuthorizationStatus {
+        var status: UNAuthorizationStatus!
+        let semaphore = DispatchSemaphore(value: 0)
+        self.getNotificationSettings { settings in
+            status = settings.authorizationStatus
+            switch status {
+            case .notDetermined:
+                break
+            default:
+                let store: ClaimsStore = globalPresentableStoreContainer.get()
+                store.send(.startClaim(from: originId))
+            }
+            semaphore.signal()
+        }
+        semaphore.wait()
```

Comment:
This would block the thread, riight? 🤔 Check `NotificationLoader`, it might help you do this by adding a loader screen in between, or we should track this in some store to make sure we don't need to synchronously await an async transaction.

Reply:
Yeah, it will block, but it is like 0.1 seconds and caused me a lot of issues. Maybe if you have some time tomorrow, we can go thought it together.